### PR TITLE
Issue #4385 - Limit new UnsupportedOperationException to direct SslContextFactory usage

### LINKED ISSUE #4385: Limit new UnsupportedOperationException to direct base class SslContextFactory usage
This change seems to have broken creation of a HTTPS connector for those using `SslContextFactory` directly instead of its `Server` subclass, like https://github.com/dropwizard/dropwizard/blob/release/1.3.x/dropwizard-jetty/src/main/java/io/dropwizard/jetty/HttpsConnectorFactory.java#L599

For a codepath that is merely "deprecated", this seems unexpected.

_Originally posted by @bentmann in https://github.com/eclipse/jetty.project/pull/4326_

### PR DESCRIPTION
**Pull request by joakime**
Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

Closes #4385

### CODE REVIEW

File: jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java [hunk 1]
```diff
@@ -1270,7 +1270,8 @@ protected KeyStore loadTrustStore(Resource resource) throws Exception
     @Deprecated
     protected X509ExtendedKeyManager newSniX509ExtendedKeyManager(X509ExtendedKeyManager keyManager)
     {
-        throw new UnsupportedOperationException("X509ExtendedKeyManager only supported on Server");
+        LOG.warn("Using Deprecated SslContextFactory implementation, SNI does not work in this context, use SslContextFactory.Server instead.");
+        return null;
```
Comment (bentmann) [inline jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java]: Looking at https://github.com/eclipse/jetty.project/blob/8a305438454c36fb0fc0db3447227a75c79ab838/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java#L1255, would it make sense to rather return the supplied `keyManager` and thereby degrade more gracefully?

Comment (peterlynch) [reply jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java]: returning null is just going from UnsupportedOperationException to NullPointerException in affected calling code. I agree a more graceful behaviour seems appropriate.

Comment (joakime) [reply jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java]: We can either return null, or throw an Exception.
Returning the unwrapped `X509ExtendedKeyManager` means broken behaviors.

**Comment by sbordet** [comment]
See comment on #4385 - I prefer an early failure than a having SNI not working properly, so I'm -1 on this change.

**Comment by bentmann** [comment]
@joakime thanks for your efforts. From the conversations so far, it's clear to me I need to look for a solution in a different place.

**Comment by joakime** [comment]
@bentmann give this PR some time for other devs to also review it.  
So far we have a neutral from me, and a -1 from Simone.

**Review by gregw — APPROVED** [review]

**Review by olamy — APPROVED** [review]
LGTM (maybe just update pr title)

**Pull request merged**